src/player.cpp

```diff
@@ -5803,10 +5803,12 @@ void player::suffer()
     }
 
     // Negative bionics effects
-    if( has_bionic( bio_dis_shock ) && one_turn_in( 2_hours ) && !has_effect( effect_narcosis ) ) {
+    if( has_bionic( bio_dis_shock ) && power_level > 9 && one_turn_in( 2_hours ) &&
+        !has_effect( effect_narcosis ) ) {
```

Comment:
This  logic seems to be wrong, now you are immune to this effect if you can keep your power lower than 9.